README.md

```diff
@@ -239,6 +239,54 @@ And finally restart your browser.
 
 If you still experience the issue, try starting your browser from terminal. If this helps, the issue is likely due to the absence of `/usr/local/bin/gpg`, follow the steps above to make sure it exists.
 
+### Configuring Browserpass on NixOS / for Nix
+
+#### On NixOS
+
+If you wish to have a stateless setup, make sure you have this in your `/etc/nixos/configuration.nix` and rebuild your system:
+
+```nix
+{ pkgs, ... }: {
+  programs.browserpass.enable = true;
+  environment.systemPackages = with pkgs; [
+    # All of these browsers will work with it
+    chromium
+    firefox
+    google-chrome
+    vivaldi
+  ];
+}
+```
+
+Note: firefox*-bin versions do _not_ work statelessly. If you require such firefox versions, use the stateful setup in the following section.
+
+#### For Nix / stateful
+
+Install browserpass native messaging host with
+
+```
+nix-env -iA nixpkgs.browserpass
+```
+
+And install the browser extension like normal. Then link the necessary files
+
+```
+# For firefox
+mkdir -p ~/.mozilla/native-messaging-hosts && \
+  ln -s ~/.nix-profile/lib/mozilla/native-messaging-hosts/com.dannyvankooten.browserpass.json ~/.mozilla/native-messaging-hosts
+# For chrome
+mkdir -p ~/.config/google-chrome/NativeMessagingHosts && \
+  ln -s ~/.nix-profile/etc/chrome-host.json ~/.config/google-chrome/NativeMessagingHosts/com.dannyvankooten.browserpass.json
+# For chromium
+mkdir -p ~/.config/chromium/NativeMessagingHosts && \
+  ln -s ~/.nix-profile/etc/chrome-host.json ~/.config/chromium/NativeMessagingHosts/com.dannyvankooten.browserpass.json
+# For vivaldi
+mkdir -p ~/.config/vivaldi/NativeMessagingHosts && \
+  ln -s ~/.nix-profile/etc/chrome-host.json ~/.config/vivaldi/NativeMessagingHosts/com.dannyvankooten.browserpass.json
```

Comment:
@Infinisil so Nix package does not put these json files into correct location and every single person has to manually symlink them after installation?

Reply:
@maximbaz Currently yes. While it is not in the spirit of Nix to require this, it can be very hard to patch browsers to look in another directory for these files. Firefox is the only one I know of that does this, and chrome might be impossible due to it being closed-source.

So on NixOS it is possible to use the `/etc` directory for chrome-based browsers and that's exactly what the `programs.browserpass.enable` option does (see https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/browserpass.nix).

In user-installations of Nix however we don't want to require users write to `/etc`, so using the user-wide paths is perfect.

There is the upcoming project [home-manager](https://github.com/rycee/home-manager) which a lot of people are using already. I might write a home-manager module for browserpass in the future which would remove the need to symlink these files manually for users of it. home-manager or something similar might become part of Nix itself in the future too.

So until this is the case, this simple manual setup is the best we can do. If you wish so, I can also modify the `install.sh` script to work properly on NixOS/Nix.

Reply:
Thanks for the detailed answer! Let's not update install.sh for now, I'm not exactly sure yet how it will look like with v3, but I'll keep Nix in mind 👍